### DISCUSSION TITLE
FaceSwap changes face at 70%

### DISCUSSION THREAD

**Discussion by Someguy342** [lllyasviel/Fooocus#2149] [Q&A] [answered]
Hi,

Firstly, I would like to say thanks for this amazing software!

Secondly, I run into the following issue when using FaceSwap and I was wondering if it happens to everyone. The generated image always significantly changes at the 70% step of the sampling process when using this feature. 21/30 for the speed setting. The preview for step 20 will look significantly more similar to the face that's being swapped than the one for step 21. I've attached an example to show what I mean. Changing the stop at values, base model and Lora does not make a difference. No refiner is used. When generating images with only prompt this doesn't happen. Haven't found any tutorials covering this topic.

Thank you so much for your time and great work.

https://github.com/lllyasviel/Fooocus/assets/158525664/83278086-374d-471d-b543-b495873de85f

**Comment by xhoxye** [comment]
Please check whether the timing of the refiner switch is 0.7 or 0.667, because the refiner will lose control after switching

**Reply by Someguy342** [reply]
Hi, thanks for answering. I used no refiner so this option wasn't visible. I've selected a refiner and tried out different "switch at" values, but the change always kept happening at 21/30. If I misunderstood and you meant the "Forced Overwrite of Refiner Switch Step" option, it was at -1, but giving it a larger value also did not impact the results.

**Comment by xhoxye** [comment, marked as answer]
FaceSwap stop at 1


https://github.com/lllyasviel/Fooocus/assets/129571231/db0c1e61-f3e5-4491-9332-254fab629770

**Reply by 007SuperMario** [reply]
I didn't understand what kind of operation you did: faceswap with only one face? 🤔 Doesn't "swap" imply 2 faces?

**Reply by mashb1t** [reply]
Faceswap swaps any detected face of the generation process with the one(s) given as reference.

**Comment by xhoxye** [comment]
I still don't know why the 21 steps lose control , because it does exist, but I tested that it disappears when the stop time is 1.
when stop at 0.9, a little girl's face becomes an adult's face at 21 steps.
30*0.9=27

https://github.com/lllyasviel/Fooocus/assets/129571231/cfd49638-58d1-4885-b80b-2d4833eba595
